app/models/event.rb

```diff
@@ -489,6 +489,37 @@ def access_method
     end
   end
 
+  def self.subj_id_check(options = {})
+    size = (options[:size] || 1000).to_i
+    cursor = (options[:cursor] || [])
+    total_errors = 0
+
+    response = Event.query(nil, source_id: "datacite-crossref,datacite-related", page: { size: 1, cursor: [] })
+    logger.info "[DoubleCheck] #{response.results.total} events for source datacite-crossref,datacite-related."
+
+    # walk through results using cursor
+    if response.results.total > 0
+      while response.results.results.length > 0 do
+        response = Event.query(nil, source_id: "datacite-crossref,datacite-related", page: { size: size, cursor: cursor })
+        break unless response.results.results.length > 0
+
+        logger.info "[DoubleCheck] DoubleCheck #{response.results.results.length}  events starting with _id #{response.results.to_a.first[:_id]}."
+        cursor = response.results.to_a.last[:sort]
+
+        # dois = response.results.results.map(&:subj_id)
+        events = response.results.results
+        events.lazy.each do | event|
+          subj_prefix = event.subj_id[/(10\.\d{4,5})/,1]
+          File.open("evens_with_double_crossref_dois.txt", "a+") do |f|
```

Comment:
Please don't use `File.open` in a container, as file storage is not persistent. Rather write to another place, e.g. via API call.

Reply:
Now i pushing the data via API to storage outside the container. However, I am still creating a file to make the whole process faster and send it over a single API call.